src/org/zaproxy/zap/extension/pscanrulesAlpha/CacheableScanner.java

```diff
@@ -407,7 +407,9 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 								lifetimeFound = true;
 								lifetimesFound++;
 								//get the portion of the string after "maxage="
-								lifetime = Long.parseLong(directiveToken.substring("max-age=".length()));
+								// Split on comma and use 0th item in case it's malformed: 
+								// Cache-Control: max-age=7776000,private
+								lifetime = Long.parseLong(directiveToken.split(",")[0].substring("max-age=".length()));
```

Comment:
I considered wrapping this in a try/catch on the off chance that a ridiculous value was encountered (exceeding `Long.MAX_VALUE`), and setting `lifetime = Long.MAX_VALUE` in the catch. Just to further ensure a NFE can't happen...think it's worth doing that? (If so I guess I could also set a relevant otherinfo message)

Reply:
Yes, please, we should not even assume that it has a number.